File: app/src/main/java/de/squiray/dailytodo/data/repository/LocalTodoRepositoryImpl.java
package de.squiray.dailytodo.data.repository;

import java.util.List;

import javax.inject.Inject;

import de.squiray.dailytodo.data.db.TodoDao;
import de.squiray.dailytodo.data.entity.mapper.TodoEntityMapper;
import de.squiray.dailytodo.domain.entity.Todo;
import de.squiray.dailytodo.domain.repository.TodoRepository;


public class LocalTodoRepositoryImpl implements TodoRepository {

    private final TodoDao todoDao;
    private final TodoEntityMapper todoEntityMapper;

    @Inject
    public LocalTodoRepositoryImpl(TodoDao todoDao,
                                   TodoEntityMapper todoEntityMapper) {
        this.todoDao = todoDao;
        this.todoEntityMapper = todoEntityMapper;
    }

    @Override
    public Todo save(Todo todo) {
        todoDao.insertTodos(todoEntityMapper.toEntity(todo));
        return todo;
    }

    @Override
    public Todo get(String todoId) {
        return todoEntityMapper.fromEntity(todoDao.get(todoId));
    }

    @Override
    public List<Todo> getAll() {
        return todoEntityMapper.fromEntities(todoDao.getAll());
    }

    @Override
    public void deleteTodo(Todo todo) {
        todoDao.deleteTodos(todoEntityMapper.toEntity(todo));
    }
}

File: app/src/main/java/de/squiray/dailytodo/data/repository/LocalTodoRepositoryImpl.kt
package de.squiray.dailytodo.data.repository

import de.squiray.dailytodo.data.db.TodoDao
import de.squiray.dailytodo.data.entity.mapper.TodoEntityMapper
import de.squiray.dailytodo.domain.entity.Todo
import de.squiray.dailytodo.domain.repository.TodoRepository
import javax.inject.Inject


class LocalTodoRepositoryImpl @Inject
constructor(private val todoDao: TodoDao,
            private val todoEntityMapper: TodoEntityMapper) : TodoRepository {

    override val all: List<Todo>
        get() = todoEntityMapper.fromEntities(todoDao.all)

    override fun save(todo: Todo): Todo {
        todoDao.insertTodos(todoEntityMapper.toEntity(todo))
        return todo
    }

    override fun get(id: String): Todo {
        return todoEntityMapper.fromEntity(todoDao.get(id))
    }

    override fun deleteTodo(todo: Todo) {
        todoDao.deleteTodos(todoEntityMapper.toEntity(todo))
    }
}
